From aed76b6b1d1d6d6b0673599695fbcb49c55961e4 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Fri, 31 Oct 2014 13:51:20 +0800
Subject: [PATCH 0439/5242] MLK-9731 ASoC: imx-hdmi-dma: audio output is noisy
 in long time playback

commit  2e0cf04bc50f78f1efe4b231a9985adec7d7725e from
https://source.codeaurora.org/external/imx/linux-imx.git

In the frame_to_bytes(), when hw_ptr*frame_bits exceed the maxmum of unsigned
long, the return value is saturated, so the appl_bytes is wrong.
This patch is to correct the usage of frame_to_bytes().

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
(cherry picked from commit 9e66132d9c96305b65aa5fa3ba8a35271a04ded9)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-hdmi-dma.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/fsl/imx-hdmi-dma.c b/sound/soc/fsl/imx-hdmi-dma.c
index 27ca56d..e3b81e5 100644
--- a/sound/soc/fsl/imx-hdmi-dma.c
+++ b/sound/soc/fsl/imx-hdmi-dma.c
@@ -458,7 +458,7 @@ static void hdmi_dma_data_copy(struct snd_pcm_substream *substream,
 	if (runtime->access != SNDRV_PCM_ACCESS_MMAP_INTERLEAVED)
 		return;
 
-	appl_bytes = frames_to_bytes(runtime, runtime->status->hw_ptr);
+	appl_bytes =  runtime->status->hw_ptr * (runtime->frame_bits / 8);
 	if (type == 'p')
 		appl_bytes += 2 * priv->period_bytes;
 	offset = appl_bytes % priv->buffer_bytes;
-- 
1.7.9.5

